### PR TITLE
feat: Drash Error Handling tutorials

### PR DESCRIPTION
**Pull request by crookse**
Add tutorials for Drash Error handling https://github.com/drashland/drash/pull/592

### CODE REVIEW

File: docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+   error: Error, // Notice this is Error -- all errors should be an extension of Error
+   request: Request, // Notice this is the native Request
+   response: Drash.Response // Notice this is the built-in Drash Response
+): void { ... }
+```
+
+A full example looks like:
+
+```typescript
+class MyErrorHandler {
```
Comment (Guergeiro) [inline docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md:59]: This needs to either implement the ErrorHandler interface or extend the ErrorHandler class.

File: docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+  public paths = ["/"];
+
+  public GET(request: Drash.Request, response: Drash.Response): void {
+    throw new BadRequestError();
+  }
+}
+
+// Create your error handler to send JSON responses
+
+class MyErrorHandler {
```
Comment (Guergeiro) [inline docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md:244]: Either implement ErrorHandler interface or extend ErrorHandler class

**Review by Guergeiro — CHANGES_REQUESTED** [review]
I also have some nitpicks. And this is something I have strong feelings about.

1. We shouldn't say *"override"* when we are just extending the class. We should say *"extend"*.
    - At most, we can point on the `catch` method that the user can use the `super.catch(...)` and "decorate" the behaviour or just ditch it entirely.
    ```ts
    class MyErrorHandler extends Drash.ErrorHandler {
      public override catch(error: Error, request: Request, response: Drash.Response) {
        console.log("Before default error handler");
        const out = super.catch(error, request, response);
        console.log("Before default error handler");
        return out;
      }
    }
    ```
2. We should say *"override"* when we are just implementing the `IErrorHandler` interface and passing this class to the server config.
3. We should teach them how to **really** *"override"* the behaviour.
    ```ts
    class MyErrorHandler implements Drash.IErrorHandler {
      public override catch(error: Error, request: Request, response: Drash.Response) {
        // handle
      }
    }
    ```

@crookse this is one of the reasons why I was so "hardcore" into this feature. This is a Proxy Pattern, that's why I forced the architecture 😉

**Comment by crookse** [comment]
> 

Is it possible to use the `override` keyword in this definition?

```typescript
class MyErrorHandler implements Drash.IErrorHandler {
  public override catch(error: Error, request: Request, response: Drash.Response) {
    // handle
  }
}
```

after reading the implementation details and docs on `override`, i see it can only be used if we `extend`, not just `implement`. if that's correct, then we don't really need `implements`, yes?

**Comment by Guergeiro** [comment]
> after reading the implementation details and docs on `override`, i see it can only be used if we `extend`, not just `implement`. if that's correct, then we don't really need `implements`, yes?

What I meant was not whether we should or shouldn't use the keyword `override`. What I meant is that there's two ways of getting an ErrorHandler, and that we should document both. One completely overrides the default implementation (implementing the interface), the other may or not override it (extending the default ErrorHandler).

**Comment by crookse** [comment]
> > after reading the implementation details and docs on `override`, i see it can only be used if we `extend`, not just `implement`. if that's correct, then we don't really need `implements`, yes?
> 
> What I meant was not whether we should or shouldn't use the keyword `override`. What I meant is that there's two ways of getting an ErrorHandler, and that we should document both. One completely overrides the default implementation (implementing the interface), the other may or not override it (extending the default ErrorHandler).

got it! making the changes now

File: docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md (hunk 2 above)
Comment (crookse) [reply docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md]: doneski: 438c6de

File: docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md (hunk 1 above)
Comment (crookse) [reply docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md]: doneski: 438c6de

**Comment by crookse** [comment]
@Guergeiro , i have the Error Handling page now going over both ways of introducing a custom error handler: 438c6de.

i also changed the term "override" to "customize" and only use the term "override" in the part where we talk about extending Drash.ErrorHandler

**Review by Guergeiro — APPROVED** [review]
Better than I was expecting! LGTM.

**Pull request merged**